### ISSUE TITLE
Tooltips are getting cut off on mobile, switch to WP component for better accessibility

### ISSUE THREAD

**Issue by elizaan36** [Automattic/woocommerce-payments#6950]
### Describe the bug
The `tooltip` component being used in WooPayments is not accessible on mobile. The text is cut off by the edge of the viewport which impacts readability. 

### To Reproduce

1. Go to Settings > Payments > WooPayments
2. Hover over any payment method that's disabled, or the fees pill in mobile view
3. See error

### Actual behavior
On smaller device sizes, the tooltips are getting cut off. When working with @mordeth on the [JCB rollout](https://github.com/Automattic/woocommerce-payments/pull/6905#issue-1833604777), it was discovered that the tooltips implemented in WCPay don't have any positioning options to account for different device sizes.

<img width="372" alt="Screen Shot 2023-08-02 at 1 48 33 PM" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/f92fcfbe-23c6-472b-9dad-dc0936298674">
<img width="374" alt="Screen Shot 2023-08-02 at 1 48 24 PM" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/55ba4de4-87e3-45ae-bcc3-c3fb3a03afeb">

### Expected behavior
As discussed in `p6riRB-8NZ#comment-9534`, we should be using the existing tooltip component in the @wordpress/components package. This will provide the ability to use dynamic positioning based on the size of the viewport as shown in this screenshot from [Storybook](https://wordpress.github.io/gutenberg/?path=/docs/components-tooltip--default).
<img width="1092" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/b821bdd2-9871-429f-b594-fb9dfd228247">

The main consideration with this update is that the tooltip persists on mouseover so the user is able to access any links included in the copy. Here are two designs to illustrate the dynamic positioning of the WP tooltip component.
<img width="777" alt="Screen Shot 2023-08-08 at 3 06 25 PM" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/cc268d97-0c50-4f40-9166-64e1f9708f93">
<img width="770" alt="Screen Shot 2023-08-08 at 3 06 34 PM" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/def6e25e-47bd-498c-9b49-46a1e39ca8f5">

Also related to #6917 cc @naman03malhotra Could this update be included as part of the #6903 scope of work?

**Comment by naman03malhotra** [comment]
I've tested it with `wordpress/Tooltip` component, and it doesn't work well. I am working on a PR which makes changes to the WCPay tooltip component to work well on the mobile.
<img width="471" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/15019298/c27a22c1-2bf6-4807-9b9d-67e2bdbc3fe1">

**Comment by naman03malhotra** [comment]
@elizaan36 I've created this PR to fix this issue https://github.com/Automattic/woocommerce-payments/pull/7515

Please have a look from the design perspective
